src/mon/Monitor.cc

```diff
@@ -1846,7 +1846,8 @@ void Monitor::handle_probe_probe(MonOpRequestRef op)
 
   dout(10) << "handle_probe_probe " << m->get_source_inst() << *m
 	   << " features " << m->get_connection()->get_features() << dendl;
-  uint64_t missing = required_features & ~m->get_connection()->get_features();
+  uint64_t missing =
```

Comment:
yeah, i think the quorum features is a superset of the required features. as the former is the intersection of connection's features and its own `CEPH_FEATURES_ALL`. if we start with a single monitor, the quorum feature is the first monitor's `CEPH_FEATURES_ALL`. and we encode the monmap using the quorum features .

so, this renders it impossible for the new monitor to join the quorum if its feature set is smaller than that of the quorum even if its feature set is able to cover the "required feature". 

so far, all looks good. but i have some concerns:

first, the comment above `Monitor::required_features` reads

> features we require of peers (based on on-disk compatset)

so, it is not valid anymore after this change. as we always have a higher bar for new joiners.

second, currently, we are using `required_features` to tell the sync client is capable enough to understand the chunks to be sent. but what if the encoded transactions (chunks) are encoded using the features not compatible with the `required_features`? put in other way, what if the sync client's feature set is a superset of host's `required_features`, but it is not able to cover the `quorum_con_features` ? i think, in that case, the sync client will fail to decode the maps.

third, if a prober fails to meet the requirement of `quorum_con_features`, the `MMonProbe:: required_features` sent to it should be `get_quorum_con_features()` not `required_features`.

but, IMHO, the second problem is more critical. so, probably, we could just use `quorum_con_features` instead of `required_features` when the sync clients want to contact the provider, to keep them out of the door until they are upgraded ? so they won't panic at seeing "corrupted" maps in the sync'ed transactions.